### PR TITLE
feat: add 'Discover more' button to banners

### PR DESCRIPTION
**Pull request by d-loose**
[Screencast from 2023-08-14 16-58-46.webm](https://github.com/ubuntu/app-store/assets/113362648/4703003d-3f67-463f-98ed-b7ef212e36bb)

The button label can be customized for each category. Since the button might need a lot of horizontal space - hide the three snap icons in the narrow layout.

### CODE REVIEW

File: lib/src/explore/explore_page.dart [hunk 1]
```diff
@@ -132,18 +145,40 @@ class _Banner extends StatelessWidget {
...
             ),
-            Row(
-              mainAxisSize: MainAxisSize.min,
-              children: snaps.map((snap) => _BannerIcon(snap: snap)).toList(),
-            ),
+            if (ResponsiveLayout.of(context).type != ResponsiveLayoutType.small)
+              Row(
+                mainAxisSize: MainAxisSize.min,
+                children: snaps.map((snap) => _BannerIcon(snap: snap)).toList(),
+              ),
```
Comment (CarlosNihelton) [inline lib/src/explore/explore_page.dart:181]: Can we animate the appearance of the row? Did Design suggested anything to smooth the appearance of those icon buttons?

File: lib/src/explore/explore_page.dart [hunk 2]
```diff
@@ -100,18 +100,31 @@ class _CategoryBanner extends ConsumerWidget {
...
+    this.buttonLabel,
+    this.onPressed,
+  });
 
   final Iterable<Snap> snaps;
   final String slogan;
+  final String? buttonLabel;
+  final VoidCallback? onPressed;
+
+  static const _kForegroundColor = Colors.white;
```
[the remark this answers is not in the document]
Comment (d-loose) [reply lib/src/explore/explore_page.dart:127]: Thanks - I completely agree. I'm planning to refactor all the private Widgets in that file into their own files, once the design is settled (there'll be tests and ~~no~~ fewer hardcoded values!). Would you be happy with a 'TODO' plan in the code? :)

**Comment by d-loose** [comment]
@CarlosNihelton, sorry - should've made clearer that we're still working on rather short and small UI iterations with lots of loose ends

**Review by CarlosNihelton — APPROVED** [review]
LGTM 👍🏽 once the TODO comments are added :D

**Comment by CarlosNihelton** [comment]
> @CarlosNihelton, sorry - should've made clearer that we're still working on rather short and small UI iterations with lots of loose ends

That's fine. Even though we may have a very complete backlog somewhere else (as GH issues or JIRA cards etc) I still consider the inline comments of great value to show others that happen to be reviewing the code (this is open source, after all) that we considered some aspects of the current implementation and avoid certain kinds of questions.

**Comment by d-loose** [comment]
Thanks a lot, that's good advice :)

**Pull request merged**